clippy_lints/src/eta_reduction.rs

```diff
@@ -101,19 +101,24 @@ fn check_clousure<'tcx>(cx: &LateContext<'tcx>, outer_receiver: Option<&Expr<'tc
     };
 
     if body.value.span.from_expansion() {
-        if body.params.is_empty() {
-            if let Some(VecArgs::Vec(&[])) = VecArgs::hir(cx, body.value) {
-                // replace `|| vec![]` with `Vec::new`
-                span_lint_and_sugg(
-                    cx,
-                    REDUNDANT_CLOSURE,
-                    expr.span,
-                    "redundant closure",
-                    "replace the closure with `Vec::new`",
-                    "std::vec::Vec::new".into(),
-                    Applicability::MachineApplicable,
-                );
-            }
+        if body.params.is_empty()
+            && let Some(VecArgs::Vec(&[])) = VecArgs::hir(cx, body.value)
+        {
+            let vec_crate = if let Some("std") = std_or_core(cx) {
```

Comment:
Better use

```suggestion
            let vec_crate = if !is_no_std_crate(cx) {
```

(and reverse the test)

Reply:
Fixed it, Thanks! https://github.com/rust-lang/rust-clippy/pull/13896/commits/41129084f224a798ee14f8b75efd4ea3774f62ea